recipe/meta.yaml

```
@@ -13,18 +13,29 @@
 {% set extension = "tar.xz" %}  # [not win]
 {% set extension = "zip" %}  # [win]
 
+{% set linux64_cuda_12 = (cuda_compiler_version or "").startswith("12") and linux64 %}
+{% set aarch64_cuda_12 = (cuda_compiler_version or "").startswith("12") and aarch64 %}
+{% set win64_cuda_12 = (cuda_compiler_version or "").startswith("12") and win64 %}
+
+{% set linux64_cuda_13 = (cuda_compiler_version or "").startswith("13") and linux64 %}
+{% set aarch64_cuda_13 = (cuda_compiler_version or "").startswith("13") and aarch64 %}
+{% set win64_cuda_13 = (cuda_compiler_version or "").startswith("13") and win64 %}
+
 package:
   name: cutensor
   version: {{ version }}
 
 source:
-  url: https://developer.download.nvidia.com/compute/cutensor/redist/libcutensor/{{ platform }}/libcutensor-{{ platform }}-{{ version }}-archive.{{ extension }}
-  sha256: db76dcbe997615ef8bc1e04c385e9a1e2016430afee7d6a242663baf17108dbf  # [linux64]
-  sha256: 484791c8967a4137c544304e7d475b1cd753d9aad5d835ae6794f68087b1175a  # [aarch64]
-  sha256: b1f5c7b399f8124ccf70e4df751e43757b7727e1028ec04a27a4a63efff06885  # [win64]
+  url: https://developer.download.nvidia.com/compute/cutensor/redist/libcutensor/{{ platform }}/libcutensor-{{ platform }}-{{ version }}_cuda{{ cuda_major }}-archive.{{ extension }}
+  sha256: 8903ea04daa32e25dda1b4d12cbdb475aefb2bb3f2317413b16c96fcadd8fd8b  # [linux64_cuda_12]
+  sha256: fdf5b10e6d09a30dae7e2956333cdb02c580a9b61d447154b7de8127ed7fa9d8  # [aarch64_cuda_12]
+  sha256: 0df80ccbfc1c607290e4ee0ef6f5400deaac674f1937f4609bff9215d1c39aef  # [win64_cuda_12]
+  sha256: 2d2768520b2d7f76a7081ddbe64c3cf61094e3e668ea296ac943c4f8435a1441  # [linux64_cuda_13]
+  sha256: 380d7b2bf8f3aebbe467225345008ccb27256d60fd1dad38665f4c7d23278a76  # [aarch64_cuda_13]
+  sha256: 643998686714108f178d1cd395aada3dd53ac89c29bc91995c42380cc4f66cec  # [win64_cuda_13]
```

So the issue that is occurring is that a selector cannot be created from a Jinja value

We can either

1. Inline the values from above into this section
2. Set a `sha` variable above that we then use once here (for [example]( https://github.com/conda-forge/libnvjpeg2k-split-feedstock/blob/2cd26c1a70fb4ae405981f0c80affa0612d2a81d/recipe/meta.yaml#L12-L15 ))

---

Guessing the former is straightforward to implement

For clarity, here is how the latter would look

```suggestion
  sha256: {{ sha }}
```